### PR TITLE
Remove deprecated usage of Mockito methods

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We used some deprecated Mockito methods.

Modifications:

- Replace deprecated method usage
- Some cleanup

Result:

No more usage of deprecated Mockito methods. Fixes [#6482].

### CODE REVIEW

File: codec-http2/src/test/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributorDependencyTreeTest.java [hunk 1]
```diff
@@ -584,8 +584,8 @@ public void sameNodeDependentShouldNotStackOverflowNorChangePrioritizableForTree
         setPriority(streamC.id(), streamA.id(), DEFAULT_PRIORITY_WEIGHT, false);
         setPriority(streamD.id(), streamA.id(), DEFAULT_PRIORITY_WEIGHT, true);
 
-        boolean[] exclusives = new boolean[] {true, false};
-        short[] weights = new short[] { DEFAULT_PRIORITY_WEIGHT/*, 100, 200, DEFAULT_PRIORITY_WEIGHT */};
+        boolean[] exclusives = {true, false};
+        short[] weights = { DEFAULT_PRIORITY_WEIGHT/*, 100, 200, DEFAULT_PRIORITY_WEIGHT */};
```
Comment (Scottmitch) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributorDependencyTreeTest.java]: whoops ... can you uncomment the values in the array? I think I did that during debug.

Comment (Scottmitch) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributorDependencyTreeTest.java]: also does our style guide care about spaces after the `{`? the line before has no spaces and this one has a space.

Comment (normanmaurer) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributorDependencyTreeTest.java]: not really I think... Let me uncomment

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (e0e8a99)

**Pull request closed without merging**